web/src/main/java/com/epam/esm/controller/CertificateRESTController.java

```diff
@@ -0,0 +1,139 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.mapper.impl.certificateMapper.CreateCertificateFromOnlyCertificateMapper;
+import com.epam.esm.model.certificate.ModelCertificate;
+import com.epam.esm.model.certificate.OnlyCertificate;
+import com.epam.esm.pagination.Pagination;
+import com.epam.esm.servise.impl.CertificateServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.Link;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/v1.1")
+public class CertificateRESTController {
+
+    private final CertificateServiceImpl service;
+    private final CreateCertificateFromOnlyCertificateMapper certificateMapper;
+
+    /**
+     * created certificate
+     * enter the following fields: name,description,duration,price
+     * dates computed automatically
+     *
+     * @param certificate the certificate
+     * @return new certificate
+     */
+    @PostMapping("/certificates")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Certificate addCertificate(@RequestBody Certificate certificate) {
```

Comment:
Good practice is to use DTO objects. See https://en.wikipedia.org/wiki/Data_transfer_object
1. All inputs and outputs in controller should be DTOs.
2.  Service layer recieves TDOs, does some operations, convert DTOs into model(like in database) objects and sends them into repository
3. Service layer also recieves model objects from repository, does some operations, and sends objects as DTOs
4. Repository layer recieves and returns model objects.
This practice r4emove tight copeling between model in database and controller inputs/outputs

Reply:
Maybe your DTOs is Model classes.
If it's true, please rename it to DTO, because "model" usually used for naming objects on repository/database layer

Reply:
rename it to DTO